nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-security/src/main/java/org/apache/nifi/web/security/saml/impl/StandardSAMLConfigurationFactory.java

```diff
@@ -0,0 +1,502 @@
+        final NiFiSAMLContextProvider contextProvider = createContextProvider(metadataManager, keyManager);
+
+        // Build the configuration instance...
+
+        return new StandardSAMLConfiguration.Builder()
+                .spEntityId(spEntityId)
+                .processor(processor)
+                .contextProvider(contextProvider)
+                .logger(createSAMLLogger(properties))
+                .webSSOProfileOptions(createWebSSOProfileOptions())
+                .webSSOProfile(createWebSSOProfile(metadataManager, processor))
+                .webSSOProfileECP(createWebSSOProfileECP(metadataManager, processor))
+                .webSSOProfileHoK(createWebSSOProfileHok(metadataManager, processor))
+                .webSSOProfileConsumer(createWebSSOProfileConsumer(metadataManager, processor))
+                .webSSOProfileHoKConsumer(createWebSSOProfileHokConsumer(metadataManager, processor))
+                .singleLogoutProfile(createSingeLogoutProfile(metadataManager, processor))
+                .metadataManager(metadataManager)
+                .extendedMetadata(extendedMetadata)
+                .backgroundTaskTimer(backgroundTaskTimer)
+                .keyManager(keyManager)
+                .authExpiration(authExpiration)
+                .identityAttributeName(identityAttributeName)
+                .groupAttributeName(groupAttributeName)
+                .requestSigningEnabled(properties.isSamlRequestSigningEnabled())
+                .wantAssertionsSigned(properties.isSamlWantAssertionsSigned())
+                .build();
+    }
+
+    private static ParserPool createParserPool() throws XMLParserException {
+        final StaticBasicParserPool parserPool = new StaticBasicParserPool();
+        parserPool.initialize();
+        return parserPool;
+    }
+
+    private static HttpClient createHttpClient(final int connectTimeout, final int readTimeout) {
+        final HttpClientParams clientParams = new HttpClientParams();
+        clientParams.setParameter(HttpConnectionParams.CONNECTION_TIMEOUT, connectTimeout);
+        clientParams.setParameter(HttpConnectionParams.SO_TIMEOUT, readTimeout);
+
+        final HttpClient httpClient = new HttpClient(clientParams);
+        return httpClient;
+    }
+
+    private static void configureCustomTLSSocketFactory(final TlsConfiguration tlsConfiguration) throws TlsException {
+        final SSLSocketFactory sslSocketFactory = SslContextFactory.createSSLSocketFactory(tlsConfiguration);
+        final ProtocolSocketFactory socketFactory = new CustomTLSProtocolSocketFactory(sslSocketFactory);
+
+        // Consider not using global registration of protocol here as it would potentially impact other uses of commons http client
+        // with in nifi-framework-nar, currently there are no other usages, see https://hc.apache.org/httpclient-3.x/sslguide.html
+        final Protocol p = new Protocol("https", socketFactory, 443);
+        Protocol.registerProtocol(p.getScheme(), p);
```

Comment:
Correct, Protocol is a class specific to commons http client, and this is registering the socket factory so that any commons http clients that make https connections will use the custom socket factory.